kernel/thread.c

```diff
@@ -287,6 +287,12 @@ void _setup_new_thread(struct k_thread *new_thread,
 	/* Any given thread has access to itself */
 	k_object_access_grant(new_thread, new_thread);
 
+	/* New threads inherit any memory domain membership by the parent */
+	if (_current->mem_domain_info.mem_domain) {
+		k_mem_domain_add_thread(_current->mem_domain_info.mem_domain,
```

Comment:
A bit curious here, isn't this supposed to be inside the 'if' block at line 296?. 
Only if the inherit flag is set we are to add the new thread to the memory domain.

Reply:
Indeed, that makes sense.

Reply:
currently the memory domain and kernel object permission mechanisms are orthogonal.
i am not sure at this time whether their policies should be combined. leaving this alone for now.